src/convertToObject.js

```diff
@@ -3,16 +3,42 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes styles string
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const KEYVALUE_SEPARATOR = ';';
+  const KEY_VALUE_SEPARATOR = ':';
+
+  const keyValueArr = filterStr(sourceString).split(KEYVALUE_SEPARATOR);
+  const resultObj = {};
+
+  keyValueArr.forEach(keyValue => {
+    let [key, value] = keyValue.split(KEY_VALUE_SEPARATOR);
+
+    if (value) {
+      key = filterStr(key); value = clearValue(filterStr(value));
+      resultObj[key] = value;
+    }
+  });
+
+  return resultObj;
+}
+
+function filterStr(str) {
```

Comment:
don't use `str` for naming...and try to give more clear name like `removeSpecificCharacters`